ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/GoToPackageDiagramHandler.java

```diff
@@ -11,20 +11,26 @@
 import org.eclipse.ui.handlers.HandlerUtil;
 import org.osate.aadl2.AadlPackage;
 import org.osate.aadl2.Element;
+import org.osate.ge.BusinessObjectContext;
 import org.osate.ge.EmfContainerProvider;
-import org.osate.ge.internal.diagram.runtime.DiagramElement;
 import org.osate.ge.internal.services.DiagramService;
 import org.osate.ge.internal.ui.editor.AgeDiagramEditor;
+import org.osate.ge.internal.util.ProjectUtil;
 
 public class GoToPackageDiagramHandler extends AbstractHandler {
 	@Override
 	public void setEnabled(final Object evaluationContext) {
 		boolean enabled = false;
-		final List<DiagramElement> selectedDiagramElements = AgeHandlerUtil
-				.getSelectedDiagramElements();
-		if (selectedDiagramElements.size() == 1) {
-			final Object selectedBo = selectedDiagramElements.get(0).getBusinessObject();
-			enabled = !(selectedBo instanceof AadlPackage) && getPackage(selectedBo) != null;
+		final List<BusinessObjectContext> selectedBusinessObjectContexts = AgeHandlerUtil
+				.getSelectedBusinessObjectContexts();
+		if (selectedBusinessObjectContexts.size() == 1) {
+			final Object selectedBo = selectedBusinessObjectContexts.get(0).getBusinessObject();
+			if(selectedBo instanceof AadlPackage) {
```

Comment:
Seems like this could be cleaned up, example:
if(!(selectedBo instanceof AadlPackage)) {
    final AadlPackage pkg = getPackage(selectedBo);
    enabled = pkg != null && ProjectUtil.getProjectForBo(pkg).isPresent();
}

Unless there is an issue I am not seeing / or just preference.